pipelined/bevy_pbr2/src/render/pbr.wgsl

```diff
@@ -432,11 +432,22 @@ struct FragmentInput {
 #ifdef VERTEX_TANGENTS
     [[location(3)]] world_tangent: vec4<f32>;
 #endif
+#ifdef VERTEX_COLORS
+#ifdef VERTEX_TANGENTS
+    [[location(4)]] world_color: vec4<f32>;
+#else
+    [[location(3)]] world_color: vec4<f32>;
+#endif
+#endif
 };
 
 [[stage(fragment)]]
 fn fragment(in: FragmentInput) -> [[location(0)]] vec4<f32> {
     var output_color: vec4<f32> = material.base_color;
+
+#ifdef VERTEX_COLORS
+    output_color = in.world_color;
```

Comment:
Doesn't it make more sense for the vertex color to be multiplied with the base color from the material, rather than overriding it?

It seems a lot more useful to me (though I admit I don't know what's the typical/standard behavior for this case / what other game engines do).

That way, the shader combines the color from the texture (if any), the color from the material (if any), and the color from the vertex (if any). This gives the most flexibility, as the user could use all of these variables to create cool graphics effects. It seems strictly more useful than what you are doing right now.

Reply:
Yep totally makes sense. I've actually moved on to another method and personally no longer use the color vertices. I set the material base color to white and then create a custom texture with the colors I need. Instead of using vertex colors I then simply use correct uv coords for the texture. Once that other pr is merged, I'll update this PR and change it to multiply instead. Great suggestion

Reply:
Which other PR?

Reply:
I think he’s referring to #3120

Reply:
I thought so but didn't see any direct reference. :) I agree it makes sense to get the flexible vertex attributes merged first and then this. I have added both of these to the rendering project board so they get prioritised as these are common problems that people hit.